pallets/gear/src/tests.rs

```diff
@@ -3706,6 +3708,234 @@ fn cascading_messages_with_value_do_not_overcharge() {
+        let gas = Gear::calculate_gas_info(
+            USER_1.into_origin(),
+            HandleKind::Handle(in_one_block),
+            pkg.encode(),
+            0,
+            true,
+        )
+        .expect("Failed to get gas spent");
+
+        (init_gas.min_limit, gas.min_limit)
+    };
+
+    let estimate_gas_for_init_and_start = || -> (u64, u64) {
+        use demo_calc_hash::sha2_256;
+        use demo_calc_hash_over_blocks::{Method, WASM_BINARY};
+
+        let block_gas_limit = BlockGasLimitOf::<Test>::get();
+
+        let init_gas = Gear::calculate_gas_info(
+            USER_1.into_origin(),
+            HandleKind::Init(WASM_BINARY.to_vec()),
+            0u64.encode(),
+            0,
+            true,
+        )
+        .expect("Failed to get gas spent");
+
+        // deploy demo-calc-hash-over-blocks
+        assert_ok!(Gear::submit_program(
+            Origin::signed(USER_1),
+            WASM_BINARY.to_vec(),
+            b"estimate over blocks".to_vec(),
+            0u64.encode(),
+            init_gas.min_limit,
+            0,
+        ));
+        let over_blocks = get_last_program_id();
+
+        run_to_next_block(None);
+
+        let (src, id, expected) = ([1; 32], sha2_256(b"estimate_over_blocks"), 0);
+
+        // Estimate start cost.
+        let start_gas_wait = Gear::calculate_gas_info(
+            USER_1.into_origin(),
+            HandleKind::Handle(over_blocks),
+            Method::Start { expected, src, id }.encode(),
+            0,
+            true,
+        )
+        .expect("Failed to get gas spent");
+
+        // Init the start message with 0 expected first.
+        assert_ok!(Gear::send_message(
+            Origin::signed(USER_1),
+            over_blocks,
+            Method::Start { src, id, expected }.encode(),
+            block_gas_limit,
+            0,
+        ));
+
+        // Estimate the gas spent on waking.
+        let start_gas_wake = Gear::calculate_gas_info(
+            USER_1.into_origin(),
+            HandleKind::Handle(over_blocks),
+            Method::Start { expected, src, id }.encode(),
+            0,
+            true,
+        )
+        .expect("Failed to get gas spent");
+
+        run_to_next_block(None);
+        SystemPallet::<Test>::reset_events();
+
+        (
+            init_gas.min_limit,
+            start_gas_wait.min_limit + start_gas_wake.min_limit,
+        )
+    };
+
+    new_test_ext().execute_with(|| {
+        use demo_calc_hash_in_one_block::{Package, WASM_BINARY};
+
+        let block_gas_limit = BlockGasLimitOf::<Test>::get();
+
+        // deply demo-calc-hash-in-one-block
+        assert_ok!(Gear::submit_program(
+            Origin::signed(USER_1),
+            WASM_BINARY.to_vec(),
+            DEFAULT_SALT.to_vec(),
+            EMPTY_PAYLOAD.to_vec(),
+            5_000_000_000,
+            0,
+        ));
+        let in_one_block = get_last_program_id();
+
+        assert!(program_exists(in_one_block.into_origin()));
```

Comment:
Importing functions is not Rust idiomatic (https://doc.rust-lang.org/stable/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html)
```suggestion
        assert!(common::program_exists(in_one_block.into_origin()));
```